src/Tools/ExternalAccess/OmniSharp/MetadataAsSource/OmniSharpMetadataAsSourceService.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.MetadataAsSource;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.OmniSharp.MetadataAsSource
+{
+    internal static class OmniSharpMetadataAsSourceService
+    {
+        /// <summary>
+        /// Generates formatted source code containing general information about the symbol's
+        /// containing assembly, and the public, protected, and protected-or-internal interface of
+        /// which the given ISymbol is or is a part of into the given document
+        /// </summary>
+        /// <param name="document">The document to generate source into</param>
+        /// <param name="symbolCompilation">The <see cref="Compilation"/> in which <paramref name="symbol"/> is resolved.</param>
+        /// <param name="symbol">The symbol to generate source for</param>
+        /// <param name="cancellationToken">To cancel document operations</param>
+        /// <returns>The updated document</returns>
+        public static Task<Document> AddSourceToAsync(Document document, Compilation symbolCompilation, ISymbol symbol, CancellationToken cancellationToken = default)
```

Comment:
💡 `CancellationToken` shouldn't be optional for non-public APIs

```suggestion
        public static Task<Document> AddSourceToAsync(Document document, Compilation symbolCompilation, ISymbol symbol, CancellationToken cancellationToken)
```

Reply:
Fair enough, but this was a copy of the existing signature :).